test/control-plane/README.md

```diff
@@ -0,0 +1,51 @@
+
+# Control-plane tests for the cilium agent
+
+The control-plane tests are integration tests that validate that the agent
+performs the correct datapath action when given Kubernetes resources as inputs.
+
+These tests are one step down from full end-to-end tests and the test cases are
+purposely written in terms of "k8s objects in, mock datapath state out" in order to
+not make any assumptions about the control-plane implementation. This approach
+caters towards:
+
+- Regression testing. Bugs in the control-plane implementation that can
+  be reproduced in a k8s environment can be converted into a test case by capturing
+  the k8s resources that describe the state of the cluster
+
+- Refactoring. Even large changes in the implementation of the control-plane
+  would not invalidate the test cases themselves. Only the "runner" implementations
+  would need to adopt to the new internal structure.
+
+
+## Running the tests
+
+The tests can be run as usual with Go test:
+
+  $ go test ./test/control-plane/...
+
+If the test case is a golden test, the golden output files can be updated
+with the '-update' flag:
+
+  $ go test ./test/control-plane/services/dual-stack -test.v -update
```

Comment:
Can we make this a make target? I'll never remember this specially if the number test features grow :see_no_evil:

Reply:
For these tests it makes more sense to run directly via `go test` as you often want to set `-test.v` or `-debug` etc. and we don't have make targets for other go tests either, so I would argue against this, but don't have a strong opinion.

Reply:
But we do have a `make integration-tests`, so I was thinking a `make tests` that would be a wrapper for `go test ./... -test.v`  and `make update-tests` for `go test ./... -update`